src/Spago/Command/Build.purs

```diff
@@ -140,56 +96,188 @@ run opts = do
+        logInfo $ "Looking for unused and undeclared transitive dependencies..."
+        maybeGraph <- Graph.runGraph globs opts.pursArgs
+        for_ maybeGraph \graph -> do
+          env <- ask
+          errors <- Graph.toImportErrors selected <$> runSpago (Record.union { graph, selected } env) Graph.checkImports
+          unless (Array.null errors) do
+            die' errors
+
+  when buildingMultiplePackages do
+    moduleMap <- liftEffect $ Ref.read duplicateModuleRef
+    let duplicateModules = Map.filter (\packages -> Array.length packages > 1) moduleMap
+    unless (Map.isEmpty duplicateModules) do
+      die $
+        ( Dodo.lines
+            [ Dodo.text $ "Detected " <> show (Map.size duplicateModules) <> " modules with the same module name across 2 or more packages defined in this workspace."
+            , duplicateModules
+                # Map.toUnfoldable
+                # Array.mapWithIndex
+                    ( \idx (Tuple m ps) ->
+                        Dodo.lines
+                          [ Dodo.text $ show (idx + 1) <> ") Module \"" <> m <> "\" was defined in the following packages:"
+                          , Dodo.indent
+                              $ Dodo.lines
+                              $ map
+                                  ( \(Tuple pkg moduleFilePath) ->
+                                      Dodo.text $ "- " <> PackageName.print pkg <> "   at path: " <> moduleFilePath
+                                  )
+                              $ Array.sort ps
+                          ]
+                    )
+                # Dodo.lines
+            ] :: Docc
+        )
 
 -- TODO: if we are building with all the packages (i.e. selected = Nothing),
 -- then we can use the graph to remove outdated modules from `output`!
 
+data SelectedPackageGlob
+  = SinglePackageGlobs WorkspacePackage
+  | AllWorkspaceGlobs (Array WorkspacePackage)
+
 type BuildGlobsOptions =
   { withTests :: Boolean
   , depsOnly :: Boolean
-  , selected :: Array WorkspacePackage
-  , dependencies :: Map PackageName Package
+  , selected :: SelectedPackageGlob
+  , dependencies :: PackageMap
   }
 
 getBuildGlobs :: BuildGlobsOptions -> Set FilePath
 getBuildGlobs { selected, dependencies, withTests, depsOnly } =
   Set.fromFoldable $ projectGlobs <> monorepoPkgGlobs <> dependencyGlobs <> [ BuildInfo.buildInfoPath ]
   where
-  -- Here we select the right globs for a monorepo setup with a bunch of packages
-  projectGlobs = join case depsOnly of
-    true -> []
-    false ->
-      -- We just select all the workspace package globs, because it's (1) intuitive and (2) backwards compatible
-      map workspacePackageGlob selected
-
   testGlobs = case withTests of
     true -> WithTestGlobs
     false -> NoTestGlobs
 
   workspacePackageGlob :: WorkspacePackage -> Array String
   workspacePackageGlob p = Config.sourceGlob testGlobs p.package.name (WorkspacePackage p)
 
+  { projectGlobs, monorepoTestGlobs } = case depsOnly of
```

Comment:
There was a comment detailing the reason why the depsOnly works that way, let's reintroduce it here